src/commands/cli/release/build.ts

```diff
@@ -0,0 +1,239 @@
+    // Works with remote branch:  "git checkout my-branch"
+    // Works with tag (detached): "git checkout 7.174.0"
+    this.exec(`git checkout ${ref}`);
+
+    const repo = await PackageRepo.create({ ux: new Ux({ jsonEnabled: this.jsonEnabled() }) });
+
+    // Get the current version for the "starting point"
+    const currentVersion = repo.package.packageJson.version;
+
+    // TODO: We might want to check and see if nextVersion exists
+    // Determine the next version based on if --patch was passed in or if it is a prerelease
+    const nextVersion = repo.package.determineNextVersion(flags.patch, flags.prerelease);
+    repo.nextVersion = nextVersion;
+
+    // Prereleases and patches need special branch prefixes to trigger GitHub Actions
+    const branchPrefix = flags.patch ? 'patch/' : flags.prerelease ? 'prerelease/' : '';
+
+    const branchName = `${branchPrefix}${nextVersion}`;
+
+    this.log(`Starting from '${ref}' (${currentVersion}) and creating branch '${branchName}'`);
+
+    // Create a new branch that matches the next version
+    this.exec(`git switch -c ${branchName}`);
+
+    if (flags.patch && pushChangesToGitHub) {
+      // Since patches can be created from any previous dist-tag or github ref,
+      // it is unlikely that we would be able to merge these into main.
+      // Before we make any changes, push this branch to use as our PR `base`.
+      // The build-patch.yml GHA will watch for merges into this branch to trigger a patch release
+      // TODO: ^ update this GHA reference once it is decided
+
+      this.exec(`git push -u origin ${branchName}`);
+    }
+
+    // bump the version in the pjson to the next version for this tag
+    this.log(`Setting the version to ${nextVersion}`);
+    repo.package.setNextVersion(nextVersion);
+    repo.package.packageJson.version = nextVersion;
+
+    if (flags.only) {
+      this.log(`bumping the following dependencies only: ${flags.only.join(', ')}`);
+      const bumped = repo.package.bumpDependencyVersions(flags.only);
+
+      if (!bumped.length) {
+        throw new SfError(
+          'No version changes made. Confirm you are passing the correct dependency and version to --only.'
+        );
+      }
+    } else {
+      // bump resolution deps
+      if (flags.resolutions) {
+        this.log('bumping resolutions in the package.json to their "latest"');
+        repo.package.bumpResolutions('latest');
+      }
+
+      // pin the pinned dependencies
+      if (flags['pinned-deps']) {
+        this.log('pinning dependencies in pinnedDependencies to "latest-rc"');
+        repo.package.pinDependencyVersions('latest-rc');
+      }
+    }
+    repo.package.writePackageJson();
+
+    this.exec('yarn install');
+    // streamline the lockfile
+    this.exec('npx yarn-deduplicate');
+
+    if (flags.snapshot) {
+      this.log('updating snapshots');
+      this.exec(`./bin/${repo.name === 'sfdx-cli' ? 'dev.sh' : 'dev'} snapshot:generate`, { silent: false });
```

Comment:
ooh, why are those different?  @WillieRuemmele fyi

Reply:
Hmmm, I actually don't know. Something to keep in mind with Parity.

Reply:
`sfdx-cli` has `dev.sh` while `cli` has only `dev`, not sure if there's any differences between one ending with an extension or not, or if it's just a change we need to make in `sfdx-cli`